packages/core/src/process/transform/convertDeploymentVertexToExecutionVertex.ts

```diff
@@ -324,5 +324,5 @@ async function resolveBytesForArtifact(
 }
 
 function assertNeverParamResult(hasParamResult: never) {
```

Comment:
I think there's a misunderstanding about what `never` means. The nerdy name for it is bottom type, and it means the absence of value (note: not undefined nor null, those are values), so if an argument has type `never` that signals to the compiler that that function will never be called.

Reply:
This is my default pattern for enforcing all values of a type in a switch statement, so if a new option is added to the type the compiler will error unless a new case is added.

Do you have a preferred pattern for achieving that without the never boilerplate above?

Reply:
I have enabled that eslint rule and changed the code to leverage it. Two things to note:

1. It won't give you the exhaustive check if there is a default option (so I removed those)
2. You are relying on the type system, so anyone who leveraged js to sneak weird data past won't trip a coherent error

I think both of those are acceptable tradeoffs for the code simplification. The code where we had more aggressive checks is deep in core, which we should control; if we get bugs in those switch checks it suggests we need better validation at the public api. Though I reserve the right to do a complete 180 if users start reporting bugs that an explicit throw would have caught.